src/lib/remark-plugins/rewrite-tutorial-links/__tests__/handle-tutorial-link.test.ts

```diff
@@ -0,0 +1,100 @@
+import { SectionOption } from 'lib/learn-client/types'
+import { handleTutorialLink } from '../utils'
+
+const TEST_TUTORIAL_SLUG = 'product/tutorial'
+const MOCK_TUTORIAL_MAP = {
+	'product/tutorial': '/product/tutorials/collection/tutorial',
+}
+
+const testEachCase = (cases: string[][]) => {
+	test.each(cases)(
+		'handleTutorialLink(%p) returns %p',
+		(input: string, expectedOutput: string) => {
+			expect(handleTutorialLink(input, MOCK_TUTORIAL_MAP)).toBe(expectedOutput)
+		}
+	)
+}
+
+describe('handleTutorialLink', () => {
+	describe('when the input is invalid', () => {
+		test.each([
+			'',
+			'/',
+			'invalid-input',
+			'still/invalid',
+			'also/invalid/',
+			'/super/invalid',
+			'another/invalid/input',
+			'/almost/valid/input',
+			'/one/more/invalid/input',
+		])('handleTutorialLink(%p) throws an error', (input: string) => {
+			expect(() => handleTutorialLink(input, {})).toThrow()
+		})
+	})
+
+	describe('when neither `search` nor `hash` are present', () => {
+		testEachCase([
+			[
+				`/tutorials/${TEST_TUTORIAL_SLUG}`,
+				`${MOCK_TUTORIAL_MAP[TEST_TUTORIAL_SLUG]}`,
+			],
```

Comment:
We should add some cases here thinking of the docs context in mind, with the absolute url - learn.hashicorp.com